packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -224,23 +254,80 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
 
   bool _debugAssertChildListLocked() => childManager.debugAssertChildListLocked();
 
+  /// Verify that the child list index is in strictly increasing order.
+  ///
+  /// This has no effect in release builds..
+  void _debugVerifyChildOrder(){
+    assert(() {
+      if (_debugChildIntegrityEnabled) {
+        if (firstChild != null) {
+          int index = indexOf(firstChild);
+          RenderBox child = childAfter(firstChild);
+          while (child != null) {
+            assert(indexOf(child) > index);
+            index = indexOf(child);
+            child = childAfter(child);
+          }
+        }
+      }
+      return true;
+    }());
+  }
+
   @override
   void insert(RenderBox child, { RenderBox after }) {
     assert(!_keepAliveBucket.containsValue(child));
     super.insert(child, after: after);
     assert(firstChild != null);
     assert(() {
-      int index = indexOf(firstChild);
-      RenderBox child = childAfter(firstChild);
-      while (child != null) {
-        assert(indexOf(child) > index);
-        index = indexOf(child);
-        child = childAfter(child);
-      }
+      _debugVerifyChildOrder();
       return true;
     }());
   }
 
+  @override
+  void move(RenderBox child, { RenderBox after }) {
+    // There are two scenarios:
+    //
+    // 1. The child is not keptAlive.
+    // The child is in childList maintained by ContainerRenderObjectMixin.
+    // We can call the super.move and update parentData with new slot.
+    //
+    // 2. The child is keptAlive.
+    // In this case, the child is no longer in the childList but might be stored in
+    // [_keepAliveBucket]. We need to update the location of the child in the bucket.
+    final SliverMultiBoxAdaptorParentData childParentData = child.parentData;
+    if (!childParentData.keptAlive) {
+      super.move(child, after: after);
+      // The following call updates the slot in the parentData.
+      childManager.didAdoptChild(child);
+      // Its slot may change even if super.move is a no op. In this case, we still
+      // want to mark as needs layout.
```

Comment:
why?

Reply:
It is possible that if we have only one child in the render tree with slot 0 and is in our viewport. 
If we move the child to slot1.
The super.move will not do anything nor markNeedsLayout since there is only one child and the position does not change in the childList, but we still need to relayout the screen since that child is no longer in slot 0.